File: Front/src/types/Requerimento.ts
import { Horario } from "./Horario";
import { Sala } from "./Sala";

export type Requerimento = {
    sala: Sala;         
    horarioInicial: Horario;
    horarioFinal: Horario;   
    data: Date;
    matricula:string;
    nome:string;   
    email:string;
    telefone:string;   
    motivoJustificativa: string;
    codigo:string;
};

File: Servicepro/src/main/java/com/back/Servicepro/models/Requerimento.java
package com.back.Servicepro.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name="TB_REQUERIMENTO")
public class Requerimento {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    private  String sala;
    private  String codigo;
    private  String horarioInicial;
    private  String horarioFinal;
    private  String data;
    private  String matricula;
    private  String nome;
    private  String email;
    private  String telefone;
    private  String motivoJustificativa;
    private  String status = "P";
    private  String nomeFunc;
    private  String matriculaFunc;
    private  String retorno="";

}
